### PR TITLE
ReaderPostServiceRemote: Updates parsing logic

### LINKED ISSUE #3325: Fix Post discussion properties
The new REST endpoint (1.1) is returning the post's *comment_count* and *comments_open* properties inside a nested dictionary. Let's update the parsing logic so that the values are correctly mapped.

```
(lldb) po [dict objectForKey:@"discussion"]
{
    "comment_count" = 5;
    "comment_status" = open;
    "comments_open" = 1;
    "ping_status" = open;
    "pings_open" = 1;
}
```

### PR DESCRIPTION
**Pull request by jleandroperez**
Fixes #3325

/cc @aerych (Thank you sir!)

### CODE REVIEW

File: WordPress/Classes/Networking/ReaderPostServiceRemote.m [hunk 1]
```diff
@@ -264,8 +265,8 @@ - (RemoteReaderPost *)formatPostDictionary:(NSDictionary *)dict
     post.blogName = [self siteNameFromPostDictionary:dict];
     post.blogDescription = [self siteDescriptionFromPostDictionary:dict];
     post.blogURL = [self siteURLFromPostDictionary:dict];
-    post.commentCount = [dict numberForKey:@"comment_count"];
-    post.commentsOpen = [[dict numberForKey:@"comments_open"] boolValue];
+    post.commentCount = [discussionDict numberForKey:@"comment_count"];
+    post.commentsOpen = [[discussionDict numberForKey:@"comments_open"] boolValue];
```
Comment (aerych) [inline WordPress/Classes/Networking/ReaderPostServiceRemote.m]: This works for the sites endpoints but the read endpoints return the information the old way and don't have a `discussion` field. 
Maybe add something like `NSDictionary *commentsDict = [dict objectForKey:@"discussion"] ? [dict dictionaryForKey:@"discussion"] : dict;` to normalize retrieval?

Comment (jleandroperez) [inline WordPress/Classes/Networking/ReaderPostServiceRemote.m]: Thanks Eric! updating right away!

**Comment by jleandroperez** [comment]
@aerych ready for a second round sir!

**Comment by aerych** [comment]
Looks good!  :shipit:

**Comment by jleandroperez** [comment]
Thank you sir!

**Pull request merged**